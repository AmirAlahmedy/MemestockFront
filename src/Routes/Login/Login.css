 .logInForm {
    width: 50%;
    min-width: 320px;
    margin: 0 auto;
    text-align: center;
    background: #fff;
    padding: 2em;
    border-radius: 4px;
}
.inputs {
    padding-top: 20px;
}
.inputs input{
    width: 60%;
    transition: width 0.4s ease-in-out;
    border-radius: 20px;
    

    outline: none;
    border: 1px solid #ccc;
    background-color: white;
    font: inherit;
    padding: 6px 10px;
    display: block;
    width: 100%;
    box-sizing: border-box;
}
.usernameInput{
    margin-bottom: 30px;
}
.registerButton {

    background-color: #0079D3;
    border: none;
    color: #fff;
    outline: none;
    cursor: pointer;
    text-align: center;
    padding: 1em;
    margin: 10px;
    font-weight: bold;
    border-radius: 3px;
    margin-left: 0;
    letter-spacing: 2px;
    text-transform: uppercase;
    
    
}
.registerButton:hover{
    background-color: #f2f2f2;
    border: 2px solid #ff4400c7;  
    color: #ff4400c7; 
}
.registerButton:focus{
    outline: none;
}
.logINFormWrapper{
    display: flex;
     flex-direction: column;
     align-items: center;
    padding-top: 5%;

}