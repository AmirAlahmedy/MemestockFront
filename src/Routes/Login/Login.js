import React, { Component } from 'react';
import './Login.css';
import Aux from '../../Components/HOC/Auxiliary';
import classes from './Login.module.css';
import { NavLink, BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import '../../Sass/styles.scss';
import Registration from '../Registration/Registration';
import axios from '../../axios-orders';


class Login extends Component {
    state = {
        forgot: " "
    }
    componentDidMount = () => {
    }
    handleSubmit(e) {
        e.preventDefault();

        let inputs = e.target.querySelectorAll('.logInForm input');
        const data = {

            Username: inputs[0].value,
            Password: inputs[1].value,

        }
        return axios.post('http://18.217.163.16/api/user/Login', data)
            .then(response => {
                if(response.data.token){
                    this.props.close();
                    this.props.finishLogin(response.data.token, data.Username);
                }else{
                    this.setState({
                        error: "*That's awkawrd, our servers are unable to process your request properly. Please try again later."
                    });
                }
            })
            .catch(error => {
                if (error.response && error.response.data && error.response.data.error) {
                    this.setState({
                        error: "*" + error.response.data.error
                    });
                } else {
                    this.setState({
                        error: "*An unknown error have occured..."
                    });
                }
            })
}
forgotPassHand(e){  
    e.preventDefault();
    let username = document.getElementsByClassName('usernameInput')[0].value;
    console.log(username);
    axios.put(`/user/ForgetPassword/${username}`)
        .then(response => {
            if(response.status === 200){
                this.setState({
                    forgot: response.data.message
                })

               console.log(response);
            }
        })
        .catch(error => {
            if(error.status === 404){
                this.setState({
                    forgot: 'User Does not exist'
                })
            }
        })
}
render(){

    let error = <p className={classes.Invalid}>{this.state.error || ""}</p>
    let frgpss = this.state.forgot != " " ?  <p className={classes.Invalid}>{"*"+this.state.forgot}</p> : null;
    return (

        <Aux>
           
            <div className="logINFormWrapper">
                <form className='logInForm' onSubmit={this.handleSubmit.bind(this)} >
                    <h1 className='logo'>LOG IN</h1>
                    {error || null}
                    {frgpss}
                    <div className='inputs'>

                        <input className="usernameInput logi"
                            type='text'
                            placeholder='User name' />

                        <input className="passInput logi"
                            type='password'
                            placeholder="Password"
                            value={this.props.password}
                            onChange={this.props.svPswrd} />

                    </div>

                    <button type="submit" className='registerButton'> Login </button>
                     <a href='#' id='frgt' onClick={this.forgotPassHand.bind(this)}>Forgot your password?</a>
                   
                </form>

            </div>
        </Aux>
    );


}
               
        }

export default Login;