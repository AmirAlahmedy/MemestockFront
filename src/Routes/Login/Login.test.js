import React from 'react';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Login from './Login';
import Aux from '../../Components/HOC/Auxiliary';
import { create } from 'react-test-renderer';

configure({adapter: new Adapter()});

describe('<Login />', () => {

  let wrapper;

  beforeEach(() => {
      wrapper = shallow(<Login />);
  });

  it('Login component renders', () => {
    expect(wrapper.find(Aux)).toHaveLength(1);
  });

  it('Login should call handleSubmit when clicked', () => {
    const spy = jest.spyOn(Login.prototype, 'handleSubmit');
    const f_wrapper = mount(<Login/>);
    f_wrapper.find('.logInForm').simulate('submit');
    expect(spy).toHaveBeenCalled() 
  });

  it('Inputs render', () => {
    expect(wrapper.find('.inputs')).toHaveLength(1);
  });

  it('Login button renders ', () => {
    expect(wrapper.find('.registerButton')).toHaveLength(1);
  });

  it('Login should call forgotPassHand when forgot password link is clicked', () => {
    const spy = jest.spyOn(Login.prototype, 'forgotPassHand');
    const f_wrapper = mount(<Login/>);
    f_wrapper.find('#frgt').simulate('click');
    expect(spy).toHaveBeenCalled() 
  });

  test("it matches the snapshot", () => {
    const component = create(<Login />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("Sends a login request", async () => {
    const component = create(<Login />);
    const instance = component.getInstance();
    await instance.handleSubmit();
    console.log(instance.state) 
  });

  it("Sends a forgot password request", async () => {
    const component = create(<Login />);
    const instance = component.getInstance();
    await instance.forgotPassHand();
    console.log(instance.state) 
  });
});