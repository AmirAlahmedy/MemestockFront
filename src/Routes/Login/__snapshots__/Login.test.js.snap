// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> it matches the snapshot 1`] = `
<div
  className="logINFormWrapper"
>
  <form
    className="logInForm"
    onSubmit={[Function]}
  >
    <h1
      className="logo"
    >
      LOG IN
    </h1>
    <p
      className="Invalid"
    >
      
    </p>
    <div
      className="inputs"
    >
      <input
        className="usernameInput logi"
        placeholder="User name"
        type="text"
      />
      <input
        className="passInput logi"
        placeholder="Password"
        type="password"
      />
    </div>
    <button
      className="registerButton"
      type="submit"
    >
       Login 
    </button>
    <a
      href="#"
      id="frgt"
      onClick={[Function]}
    >
      Forgot your password?
    </a>
  </form>
</div>
`;
